src/pip/_internal/configuration.py

```diff
@@ -330,35 +330,38 @@ def get_environ_vars(self) -> Iterable[Tuple[str, str]]:
     def iter_config_files(self) -> Iterable[Tuple[Kind, List[str]]]:
```

Comment:
I've not looked closely but this isn't merely changing the documentation but _also_ the implementation. For a PR titled "Fix documentation", that seems incorrect.

Could you elaborate on why fixing the documentation to match existing behaviour requires a change in the implementation? I don't think that's necessary and I think we should leave the implementation as-is and document what's implemented.

Reply:
The only change to the actual implementation is changing the order of files in `iter_config_files` (and just moving the env file last in this case). I sort of felt this was morally a documentation change since I believe the only thing this actually does is change the order that things are displayed to users. I could split this into another PR or change the title of this one to something like `Fix ergonomics of PIP_CONFIG_FILE`.

Reply:
@pradyunsg any additional thoughts?

Reply:
bump